### PR TITLE
perf(angular): upgrade to Angular 1.7.7

### LINKED ISSUE #766: Support for AngularJS 1.7 LTS
At this moment angular-patternfly supports [1] only Angular 1.5.*

It would be very useful to have the support also for AngularJS 1.7 LTS that will be supported until June 30, 2021 [2] 

Thanks

[1] https://github.com/patternfly/angular-patternfly/blob/master/bower.json#L40
[2] https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c

### PR DESCRIPTION
**Pull request by redallen**
## Description
Closes #766 .

## PR Checklist

- [x] Unit tests are included
- [x] Screenshots are attached (if there are visual changes in the UI)
- [x] A Designer (@beanh66) is assigned as a reviewer (if there are visual changes in the UI)
- [x] A CSS rep (@cshinn) is assigned as a reviewer (if there are visual changes in the UI)
- [x] Fix pf-bootstrap-select directive unit tests
- [x] Fix pfCanvas and pfCanvasEditor
- [x] Fix example script urls (I trust `<script src="http://localhost:8000/grunt-scripts/jquery.js"></script>` will become `<script src="https://www.patternfly.org/angular-patternfly/grunt-scripts/jquery.js"></script>` somewhere in the deploy)

## Additional notes
All urls in the app have changed to use `#!` instead of `#`. See the [Angular 1.6 changelog](https://docs.angularjs.org/guide/migration#commit-aa077e8). As a result, we need to upgrade to [grunt-uidocs-generator](https://www.npmjs.com/package/grunt-uidocs-generator).

### CODE REVIEW

File: src/canvas-view/examples/canvas.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**
  * @ngdoc directive
- * @name patternfly.canvas.component:pfCanvas
+ * @name patternfly.canvas.directive:pfCanvas
```
Comment (dtaylor113) [inline src/canvas-view/examples/canvas.js]: Hi, why the switch from `component` to `directive`?

File: src/card/aggregate-status/aggregate-status-card.component.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /**
  * @ngdoc directive
- * @name patternfly.card.component:pfAggregateStatusCard
+ * @name patternfly.card.directive:pfAggregateStatusCard
```
Comment (dtaylor113) [inline src/card/aggregate-status/aggregate-status-card.component.js]: Why the change to `directive`?  pfAggregateStatusCard does appear to be a component:
```
angular.module( 'patternfly.card' ).component('pfAggregateStatusCard', {
```

File: src/canvas-view/examples/canvas.js (hunk 1 above)
Comment (redallen) [reply src/canvas-view/examples/canvas.js]: Needed for grunt-ui-docs to even show the component on the generated doc webpage.

File: src/card/aggregate-status/aggregate-status-card.component.js (hunk 2 above)
Comment (redallen) [reply src/card/aggregate-status/aggregate-status-card.component.js]: Needed for grunt-ui-docs to even show the component on the generated doc webpage.

File: src/canvas-view/examples/canvas.js (hunk 1 above)
Comment (dtaylor113) [reply src/canvas-view/examples/canvas.js]: Ok, yeah, I thought it would be something like that. ok

File: src/card/aggregate-status/aggregate-status-card.component.js (hunk 2 above)
Comment (dgutride) [reply src/card/aggregate-status/aggregate-status-card.component.js]: Yeah - something changed after angular 1.6 that requires the name match directive now (and components aren't natively supported by grunt-ui-docs or ngdocs)

Comment (dtaylor113) [reply src/card/aggregate-status/aggregate-status-card.component.js]: Ok, thought it was something along those lines

**Comment by dtaylor113** [comment]
Hi @redallen, great work!

I downloaded this branch and ran though most uidoc examples: charts, wizards, toolbar, table, nav, notification drawer, filter, etc...  Everything ran as expected!

Couple of issues:

1. You have to squash and change the commit msg(s) to a single: "perf(angular): upgrade to Angular 1.7.7" (not the PR description).  Assuming we are going with a Major change (due to # files changed, as opposed to 'breaking changes'.)

2. I think I have to manually merge this PR, looks like coveralls is failing, not sure why it would though?  

Thanks,
- Dave

**Comment by redallen** [comment]
I just squashed the commit and force pushed. Coveralls was failing because test coverage went down 0.8% (not sure why...)

**Comment by dtaylor113** [comment]
Also, pls note that the 'npm publish' step will most likely fail after merge.  I will have to create a PR to manually update the version number, similar to #770

**Comment by dtaylor113** [comment]
@redallen, thanks.  Would like to get @dgutride to sign off and approve the "perf(angular): upgrade to Angular 1.7.7" major version decision.  Assuming we are going with a Major change (due to # files changed, as opposed to 'breaking changes'.) -thanks

**Comment by jeff-phillips-18** [comment]
Not sure we need to do a Major release only because of the number of files. If it's non-breaking, there shouldn't be a need to bump the major version.

**Comment by dtaylor113** [comment]
> Not sure we need to do a Major release only because of the number of files. If it's non-breaking, there shouldn't be a need to bump the major version.

I 'somewhat' agree, but there are a lot of source code changes and dependency version updates.  I'm for minimizing risk to existing projects who don't necessarily need angular 1.7.7.  Even though all unit tests are passing, I'm concerned dependency version updates may affect existing customers in ways we aren't testing for.
